backend/app/core/scraper_service.py

```
@@ -0,0 +1,354 @@
+                        saved_events=result.saved_events,
+                        processing_time=result.processing_time,
+                        source=result.source,
+                        errors=result.errors if result.errors else None
+                    )
+                    
+                except Exception as e:
+                    self.error_manager.record_operation(success=False)
+                    raise HTTPException(
+                        status_code=500,
+                        detail=f"Scraping failed for {source}: {str(e)}"
+                    )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)